ConsoleApp1/Poll.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApp1
+{
+    internal class Poll
```

Comment:
you almost everywhere check that poll is complex or single, maybe you need to create 2 classes for each poll type?